synfig-core/src/synfig/blur.cpp

```diff
@@ -54,8 +54,8 @@ using namespace synfig;
 # if __has_cpp_attribute(fallthrough)
 #  define fallthrough__ [[fallthrough]]
 # elif __has_cpp_attribute(noreturn)
-[[noreturn]] void fake_fallthrough___() {}
```

Comment:
`# elif __has_cpp_attribute(noreturn)`
Thinking about it now, we should remove those lines: the `if` guard is about `noreturn` attribute as a trick to avoid the fallthrough warning, and we are not using the `[[noreturn]]` attribute ;)

Reply:
I suggest more universal way, then:
```
#if __GNUC__ >= 7
#  define fallthrough__ __attribute__((fallthrough))
#elif __clang__
#  define fallthrough__ [[clang::fallthrough]]
#else
#  define fallthrough__ ((void)0)
#endif
```

Any suggestions?

Reply:
> I suggest more universal way, then

In the "last" `else` branch?



> P.S. BTW maybe it's time to switch to c++17? :)

I don't know lol. Since what debian version - or since what year - we are supporting or intend to support?
Do it now or later (when?)?

Reply:
> In the "last" else branch?

I mean this part:
```
#elif __clang__
#  define fallthrough__ [[clang::fallthrough]]
```
Now it works correctly with clang too.
In the last else branch I switched to `((void)0)` as more common way for no op which does nothing.
This approach used in `assert.h` or boost for example. I can revert it back if you want.

> I don't know lol. Since what debian version - or since what year - we are supporting or intend to support?
> Do it now or later (when?)?

I've thought about it and think we should stay with C++11, at least for this release.
macOS 10.9 clang does not support C++17, so we should abandon it if we move to C++17.

Reply:
What about mixing them?
```c++
#if defined(__has_cpp_attribute)
# if __has_cpp_attribute(fallthrough)
#  define fallthrough__ [[fallthrough]]
# endif
#endif

#ifndef fallthrough__
# if __GNUC__ >= 7
#  define fallthrough__ __attribute__((fallthrough))
# elif __clang__
#  define fallthrough__ [[clang::fallthrough]]
# else
#  define fallthrough__ ((void)0)
# endif
#endif
```

Reply:
Agree.